components/script/script_thread.rs

```diff
@@ -3566,11 +3566,13 @@ impl ScriptThread {
                 // We might have to reset the anchor state
                 if !state_already_changed {
                     if let Some(target) = prev_mouse_over_target {
-                        if let Some(_) = target
+                        if  Some(
+                            target
                             .upcast::<Node>()
                             .inclusive_ancestors(ShadowIncluding::No)
                             .filter_map(DomRoot::downcast::<HTMLAnchorElement>)
                             .next()
+                        ).is_some()
```

Comment:
This will always be true since the change tests `Some(...).is_some()` , right? If you want to get rid of of the `let` binding this should just be:

```rust
                   if target.upcast::<Node>()
                            .inclusive_ancestors(ShadowIncluding::No)
                            .filter_map(DomRoot::downcast::<HTMLAnchorElement>)
                            .next()
                            .is_some() {
                                  ....
                            }
```

Reply:
Wow, thank youuuu!

Reply:
I'll improve.